source/common/upstream/outlier_detection_impl.h

```diff
@@ -43,16 +81,43 @@ class OutlierDetectorHostSinkImpl : public OutlierDetectorHostSink {
  */
 class OutlierDetectorImpl : public OutlierDetector {
 public:
-  OutlierDetectorImpl(Cluster& cluster, Event::Dispatcher& dispatcher);
+  void onConsecutive5xx(HostPtr host);
+  Runtime::Loader& runtime() { return runtime_; }
 
   // Upstream::OutlierDetector
   void addChangedStateCb(ChangeStateCb cb) override { callbacks_.push_back(cb); }
 
+protected:
+  OutlierDetectorImpl(Cluster& cluster, Event::Dispatcher& dispatcher, Runtime::Loader& runtime,
+                      Stats::Store& stats, SystemTimeSource& time_source);
+
 private:
   void addHostSink(HostPtr host);
+  void armIntervalTimer();
```

Comment:
nit: call that in a way to easy get that this is associated with host unejection